presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ColumnRangesSystemTable.java

```diff
@@ -0,0 +1,188 @@
+import static com.facebook.presto.raptor.metadata.DatabaseShardManager.shardIndexTable;
+import static com.facebook.presto.raptor.util.DatabaseUtil.metadataError;
+import static com.facebook.presto.spi.SystemTable.Distribution.SINGLE_COORDINATOR;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class ColumnRangesSystemTable
+        implements SystemTable
+{
+    private static final String MIN_COLUMN_SUFFIX = "_min";
+    private static final String MAX_COLUMN_SUFFIX = "_max";
+    private static final String COLUMN_RANGES_TABLE_SUFFIX = "$column_ranges";
+
+    private final IDBI dbi;
+    private final RaptorTableHandle sourceTable;
+    private final List<TableColumn> indexedRaptorColumns;
+    private final ConnectorTableMetadata tableMetadata;
+
+    public ColumnRangesSystemTable(RaptorTableHandle sourceTable, IDBI dbi)
+    {
+        this.sourceTable = requireNonNull(sourceTable, "sourceTable is null");
+        this.dbi = requireNonNull(dbi, "dbi is null");
+
+        this.indexedRaptorColumns = dbi.onDemand(MetadataDao.class).listTableColumns(sourceTable.getTableId()).stream()
+                .filter(column -> isIndexedType(column.getDataType()))
+                .collect(toImmutableList());
+        List<ColumnMetadata> systemTableColumns = this.indexedRaptorColumns.stream()
+                .flatMap(column -> Stream.of(
+                        new ColumnMetadata(column.getColumnName() + MIN_COLUMN_SUFFIX, column.getDataType(), null, false),
+                        new ColumnMetadata(column.getColumnName() + MAX_COLUMN_SUFFIX, column.getDataType(), null, false)))
+                .collect(toImmutableList());
+        SchemaTableName tableName = new SchemaTableName(sourceTable.getSchemaName(), sourceTable.getTableName() + COLUMN_RANGES_TABLE_SUFFIX);
+        this.tableMetadata = new ConnectorTableMetadata(tableName, systemTableColumns);
+    }
+
+    public static Optional<SchemaTableName> tryGetSourceTable(SchemaTableName tableName)
+    {
+        if (isColumnRangesTable(tableName)) {
+            return Optional.of(new SchemaTableName(
+                tableName.getSchemaName(),
+                    tableName.getTableName().substring(0, tableName.getTableName().length() - COLUMN_RANGES_TABLE_SUFFIX.length())));
+        }
+        return Optional.empty();
+    }
+
+    private static boolean isColumnRangesTable(SchemaTableName tableName)
+    {
+        return tableName.getTableName().endsWith(COLUMN_RANGES_TABLE_SUFFIX) && tableName.getTableName().length() > COLUMN_RANGES_TABLE_SUFFIX.length();
```

Comment:
Instead of comparing length, what if we make this
```java
return tableName.getTableName().endsWith(COLUMN_RANGES_TABLE_SUFFIX) &&
        !tableName.getTableName().equals(COLUMN_RANGES_TABLE_SUFFIX);